### ISSUE TITLE
Clippy complains

### ISSUE THREAD

**Issue by stappersg** [ebfull/pcap#157]
Hi,

```
clippy-driver  rustc --crate-name pcap --edition=2018 src/lib.rs
```

reports errors.

Probably a known issue, but now there is an [issue](https://github.com/ebfull/pcap/issues) about it.

**Comment by stappersg** [comment]
Complain from `clippy` as found for #156
```
 error: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
  --> src/stream.rs:47:5
   |
47 |     fn decode<'a>(&mut self, packet: Packet<'a>) -> Result<Self::Type, Error>;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::needless-lifetimes` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
```

In my `rustup` environment I get complians from _clippy_ about  `libc`, but nothing about `lifetimes` ...

**Comment by sekineh** [comment]
stable = 1.48 OK
https://github.com/ebfull/pcap/actions/runs/453647534

stable = 1.49 Fails
https://github.com/ebfull/pcap/actions/runs/455364191

**Comment by Wojtek242** [comment]
Fixed in #159
